dex/src/main/scala/com/wavesplatform/dex/Application.scala

```diff
@@ -325,47 +325,69 @@ class Application(settings: MatcherSettings, config: Config)(implicit val actorS
       blacklistedAddresses,
       hasMatcherAccountScript
     )(_)
-
-    new MatcherApiRoute(
-      pairBuilder,
-      matcherPublicKey,
-      config,
-      orderBookDirectoryActorRef,
-      addressDirectoryRef,
-      wavesBlockchainAsyncClient.status(),
-      matcherQueue.store,
-      p => Option(orderBooks.get()) flatMap (_ get p),
-      orderBookHttpInfo,
-      getActualTickSize = { assetPair =>
-        getAndCacheDecimals(assetPair.amountAsset)
-          .product(getAndCacheDecimals(assetPair.priceAsset))
-          .map { case (amountAssetDecimals, priceAssetDecimals) =>
-            matchingRulesCache.getDenormalizedRuleForNextOrder(assetPair, lastProcessedOffset, amountAssetDecimals, priceAssetDecimals).tickSize
-          }
-      },
-      orderValidation,
-      settings,
-      () => status,
-      orderDb,
-      () => lastProcessedOffset,
-      () => matcherQueue.lastOffset,
-      ExchangeTransactionCreator.getAdditionalFeeForScript(hasMatcherAccountScript),
-      maybeApiKeyHash,
-      rateCache,
-      validatedAllowedOrderVersions = () => {
-        Future
-          .sequence {
-            settings.allowedOrderVersions.map(OrderValidator.checkOrderVersion(_, wavesBlockchainAsyncClient.isFeatureActivated).value)
-          }
-          .map(_.collect { case Right(version) => version })
-      },
-      () => orderFeeSettingsCache.getSettingsForOffset(lastProcessedOffset + 1),
-      externalClientDirectoryRef,
-      getAndCacheDescription
-    )
+    new PlaceRoute(pairBuilder, addressDirectoryRef, orderValidation, settings, () => status, maybeApiKeyHash)
   }
 
-  private val httpApiRouteV1 = MatcherApiRouteV1(pairBuilder, orderBookHttpInfo, () => status, maybeApiKeyHash)
+  private val cancelRoute = new CancelRoute(pairBuilder, addressDirectoryRef, () => status, orderDb, maybeApiKeyHash, settings)
```

Comment:
1. In all routes except `MarketRoute`, we use only `actorResponseTimeout` from the `settings` in routes:
    ```scala
    implicit private val timeout: Timeout = matcherSettings.actorResponseTimeout
    ```
    Let's replace `settings` with `responseTimeout: FiniteDuration`
2. We always pass settings as the first argument. So, let's move `responseTimeout`